PowerEditor/src/WinControls/DockingWnd/DockingCont.cpp

```diff
@@ -952,18 +974,18 @@ LRESULT DockingCont::runProcTab(HWND hwnd, UINT Message, WPARAM wParam, LPARAM l
 			tcItem.mask		= TCIF_PARAM;
 			::SendMessage(hwnd, TCM_GETITEM, iItem, reinterpret_cast<LPARAM>(&tcItem));
 			if (!tcItem.lParam)
-				return FALSE;
+				return 1;
 
 			toolTip.init(_hInst, hwnd);
 			toolTip.Show(rc, ((tTbData*)tcItem.lParam)->pszName, info.pt.x, info.pt.y + 20);
-			return TRUE;
+			return 0;
 		}
 
 		case WM_MOUSELEAVE:
 		{
 			toolTip.destroy();
 			_bTabTTHover = FALSE;
-			return TRUE;
+			return 0;

```

Comment:
The above return values are changed:
1. from BOOL to int
2. from TRUE to 0 & FALSE to 1 - they are reversed

Is it about unintentional error?

Reply:
@donho 
we are subclassing to window procedure which use "reversed" values to dialog procedure

Reply:
@ozone10 
Could you elaborate please?

Reply:
@donho 
[SetWindowSubclass](https://learn.microsoft.com/en-us/windows/win32/api/commctrl/nf-commctrl-setwindowsubclass) will install window procedure, which is different to dialog box procedure. 
Dialog box procedure returns for most messages TRUE if it processes a message or FALSE if it does not.

[Control-Color Messages](https://learn.microsoft.com/en-us/windows/win32/dlgbox/dlgbox-programming-considerations#control-color-messages) handling in dialog box procedure is main hint to differentiate procedures.

When returning FALSE from not subclassed dialog, default coloring is used (message is ignored), but if subclassed, controls will be colored in black since FALSE == NULL, meaning returned HBRUSH is nullptr which is invalid.

Reply:
> [SetWindowSubclass](https://learn.microsoft.com/en-us/windows/win32/api/commctrl/nf-commctrl-setwindowsubclass) will install window procedure, which is different to dialog box procedure.
Dialog box procedure returns for most messages TRUE if it processes a message or FALSE if it does not.

In this case, I prefer to not use SetWindowSubclass - Let's not modify the return value and keep the legacy windows subclass way.